### ISSUE TITLE
Change assessment/improve lists presentation

### ISSUE THREAD

**Issue by ifoche** [EyeSeeTea/malariapp#758]

**Comment by ifoche** [comment]
There are some problems related to lists like #734, derived from the way at this moment we're giving the impression to the user that the lists consist in 2 different elements (Header and row), while in reality, there is just one row element directly attached to the survey object that is showing or hiding the OrgUnit title depending on the precedent of following elements. As the client asked us to make the row height bigger for the cases of rows are hiding their title, it has become clear that we need to reformulate the implementation of the lists, to include the presentation/content separation that we have in planning tab. It would be perfect to refactor a bit those classes (PlannedHeader, PlannedItem, PlannedSurvey, etc) to make them more abstract and extend them from every tab, creating also a ImproveHeader, AssessHeader, ImproveItem, AssessItem, etc.

This task is a bit challenging, as it will have a lot of collateral modifications (surveys filters, events to access each survey, etc). Please be careful when doing it, checking and rechecking. Maybe you can identify some tests while doing it

**Comment by ifoche** [comment]
@ivanarrizabalaga please don't start this task before talking to me. This task has reduced its priority, and another must be addressed before.

**Comment by ivanarrizabalaga** [comment]
@ifoche fine!



[image: http://]

Ivan Arrizabalaga
[image: http://]about.me/ivan-arrizabalaga
<http://about.me/ivan-arrizabalaga>

**Comment by ivanarrizabalaga** [comment]
- [ ] Refactor Sent/Unsent adapters
- [ ] Solve 'phantom' orgunit header issue

**Comment by xurxodev** [comment]
@ifoche this issue has a pull request and has ToTest, H1.3 labels but the last commit is of December 2016 and exists many conflicts, what do we do with this issue?

**Comment by idelcano** [comment]
@ifoche  @xurxodev  its my fault, I used the branch https://github.com/EyeSeeTea/malariapp/tree/refactor_adapters_merged to not lose anything and i forgot to update the PR branch.

**Comment by ifoche** [comment]
@idelcano I guess you need to create a new PR referring the issue from that branch and close this one?

**Comment by idelcano** [comment]
@ifoche  yes, i closed the older PR and i created a new PR with the new branch. I did a new merge with development branch (to update the conflicts with the model changes) @xurxodev  you can review again.
